docs/topics/hetzner-baremetal.md

```diff
@@ -0,0 +1,377 @@
+export HETZNER_SSH_PRIV_PATH="<YOUR-SSH-PRIVATE-PATH>"
+```
+
+- HCLOUD_TOKEN: The project where your cluster will be placed. You have to get a token from your HCloud Project.
+- HETZNER_ROBOT_USER: The User you have defined in Robot under settings/web.
+- HETZNER_ROBOT_PASSWORD: The Robot Password you have set in Robot under settings/web.
+- HETZNER_SSH_PUB_PATH: The Path to your generated Public SSH Key.
+- HETZNER_SSH_PRIV_PATH: The Path to your generated Private SSH Key. This is needed because CAPH uses this key to provision the node in Hetzner Dedicated.
+
+```shell
+kubectl create secret generic hetzner --from-literal=hcloud=$HCLOUD_TOKEN --from-literal=robot-user=$HETZNER_ROBOT_USER --from-literal=robot-password=$HETZNER_ROBOT_PASSWORD
+
+kubectl create secret generic robot-ssh --from-literal=sshkey-name=test --from-file=ssh-privatekey=$HETZNER_SSH_PRIV_PATH --from-file=ssh-publickey=$HETZNER_SSH_PUB_PATH
+```
+
+> NOTE: sshkey-name should must match the name that is present in hetzner otherwise the controller will not know how to reach the machine.
+
+Patch the created secrets so that they get automatically moved to the target cluster later. The following command helps you do that:
+
+```shell
+kubectl patch secret hetzner -p '{"metadata":{"labels":{"clusterctl.cluster.x-k8s.io/move":""}}}'
+kubectl patch secret robot-ssh -p '{"metadata":{"labels":{"clusterctl.cluster.x-k8s.io/move":""}}}'
+```
+
+The secret name and the tokens can also be customized in the cluster template.
+
+
+### Applying Cluster Manifest
+Since we have already created secret in hetzner robot, hcloud and ssh-keys as secret in management cluster, we can now apply the cluster.
+```bash
+kubectl apply -f my-cluster.yaml
+```
+
+```bash
+$ kubectl apply -f my-cluster.yaml
+kubeadmconfigtemplate.bootstrap.cluster.x-k8s.io/my-cluster-md-0 created
+kubeadmconfigtemplate.bootstrap.cluster.x-k8s.io/my-cluster-md-1 created
+cluster.cluster.x-k8s.io/my-cluster created
+machinedeployment.cluster.x-k8s.io/my-cluster-md-0 created
+machinedeployment.cluster.x-k8s.io/my-cluster-md-1 created
+machinehealthcheck.cluster.x-k8s.io/my-cluster-control-plane-unhealthy-5m created
+machinehealthcheck.cluster.x-k8s.io/my-cluster-md-0-unhealthy-5m created
+machinehealthcheck.cluster.x-k8s.io/my-cluster-md-1-unhealthy-5m created
+kubeadmcontrolplane.controlplane.cluster.x-k8s.io/my-cluster-control-plane created
+hcloudmachinetemplate.infrastructure.cluster.x-k8s.io/my-cluster-control-plane created
+hcloudmachinetemplate.infrastructure.cluster.x-k8s.io/my-cluster-md-0 created
+hcloudremediationtemplate.infrastructure.cluster.x-k8s.io/control-plane-remediation-request created
+hcloudremediationtemplate.infrastructure.cluster.x-k8s.io/worker-remediation-request created
+hetznerbaremetalmachinetemplate.infrastructure.cluster.x-k8s.io/my-cluster-md-1 created
+hetznercluster.infrastructure.cluster.x-k8s.io/my-cluster created
+```
+
+### Creating Object Manifests
```

Comment:
It doesn't make sense IMO to do this after creating the cluster.. I think we should have the following order:
1. Preparing provider
2. Creating management cluster
3. Creating secrets in management cluster
4. Creating host objects in mgt cluster
5. Creating workload cluster